packages/stream-plugin/src/index.ts

```diff
@@ -44,46 +45,67 @@ export interface ActionCtx<T extends { [K in keyof T]: Stream<any> }> {
   [valueSymbol]: Partial<UnStreams<T>>;
 }
 
-const init = <T extends { [K in keyof T]: Stream<any> }>(
-  _config: {},
-  universe: Universe<T>,
-) => {
-  universe.streams = {};
-};
+export interface Event<T extends { [K in keyof T]: Stream<any> }> {
+  streams: { [K in keyof T]?: T[K] | (typeof deletedSymbol) };
+}
 
-const prepareActionCtx = <T extends { [K in keyof T]: Stream<any> }>(
-  state: State<T>,
-  streamUpdate: PluginAction<ActionCtx<T>, [keyof T, any]>,
-) => ({
+const prepareActionCtx = <T extends { [K in keyof T]: Stream<any> }>({
   ctx,
   universe,
+  event,
 }: {
-  ctx: PluginActionCtx<ActionCtx<T>, Universe<T>> & ActionCtx<T>;
+  ctx: ActionCtx<T>;
   universe: Universe<T>;
+  event: Event<T>;
 }) => {
+  // For the streamUpdate action
   ctx[valueSymbol] = universe.streams;
-  ctx.streams = new Proxy(state.streams, {
-    set: (obj, key: keyof T, value) => {
-      const cb = (value: any) => {
-        ctx.rootDispatch(streamUpdate)(key, value);
-      };
-      state.states[key] = value.subscribe(cb);
+
+  // For user actions
+  event.streams = {};
+  ctx.streams = new Proxy(event.streams as Partial<T>, {
+    get: () => undefined,
+    set: <K extends keyof T>(obj, key: K, value: T[K]) => {
       obj[key] = value;
       return true;
     },
     deleteProperty: (obj, key: keyof T) => {
-      const streamState = state.states[key];
-      if (streamState != null) {
-        streamState.unsubscribe();
-      }
+      // Typescript doesn't know about proxies (typescript#20846)
```

Comment:
Maybe put link to issue here instead of just the number as it is more clear where this comes from.

Reply:
Done.